huaweicloud/services/as/resource_huaweicloud_as_notification.go

```diff
@@ -0,0 +1,241 @@
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceAsNotification() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAsNotificationPut,
+		UpdateContext: resourceAsNotificationPut,
+		ReadContext:   resourceAsNotificationRead,
+		DeleteContext: resourceAsNotificationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAsNotificationImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"scaling_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the AS group ID.`,
+			},
+			"topic_urn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the unique topic URN of the SMN.`,
+			},
+			"events": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"SCALING_UP", "SCALING_UP_FAIL", "SCALING_DOWN", "SCALING_DOWN_FAIL", "SCALING_GROUP_ABNORMAL",
+					}, false),
+				},
+				Description: `Specifies the topic scene of AS group.`,
+			},
+			"topic_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The topic name in SMN.`,
+			},
+		},
+	}
+}
+
+func resourceAsNotificationPut(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// putASNotification: put an AS notification.
+	var (
+		putASNotificationHttpUrl = "autoscaling-api/v1/{project_id}/scaling_notification/{scaling_group_id}"
+		putASNotificationProduct = "autoscaling"
+	)
+	putASNotificationClient, err := cfg.NewServiceClient(putASNotificationProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating AutoScaling Client: %s", err)
+	}
+
+	putASNotificationPath := putASNotificationClient.Endpoint + putASNotificationHttpUrl
+	putASNotificationPath = strings.ReplaceAll(putASNotificationPath, "{project_id}",
+		putASNotificationClient.ProjectID)
+	putASNotificationPath = strings.ReplaceAll(putASNotificationPath, "{scaling_group_id}",
+		fmt.Sprintf("%v", d.Get("scaling_group_id")))
+
+	putASNotificationOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	putASNotificationOpt.JSONBody = buildCreateOrUpdateASNotificationBodyParams(d)
+	_, err = putASNotificationClient.Request("PUT", putASNotificationPath, &putASNotificationOpt)
+	if err != nil {
+		return diag.Errorf("error creating or updating AS notification: %s", err)
+	}
+
+	topicUrn := d.Get("topic_urn").(string)
+	d.SetId(topicUrn)
```

Comment:
can the user associate a different as_group to the same topic Urn?

Reply:
I have tried it, different as_groups can use the same topic Urn.